### ISSUE TITLE
Refactor messages_by_contact_date view to reduce more

### ISSUE THREAD

**Issue by garethbowen** [medic/medic-webapp#4284]
The `medic-client/messages_by_contact_date` view is designed to return an object which contains all the information needed for the message list reduced to show only the last message from each contact. As the amount of information needed has increased so has the emitted value which means sometimes it crashes because the reduce isn't reducing the size of the object as much as expected (#4101). To get around this we introduced some message truncation so long messages wouldn't break, but this is buggy with unicode characters and also makes the view code unnecessarily complicated.

Refactor this to return just the `reported_date` as the value and reduce to find the latest message. Then on the client side get all the IDs of the latest messages and fetch the docs in a separate request. This is slightly worse client side performance (two requests instead of one) but much easier on the server view generator and will always reduce the output enough to avoid the error. It also means we don't need to do message truncation we can just rely on the css ellipsis overflow.

**Comment by garethbowen** [comment]
This is probably blocked by #3826

**Comment by newtewt** [comment]
@vimemo  can you share some things for AT on this. I would assume checking message functionality works but I don't really know all of the message functionality.

**Comment by vimemo** [comment]
@newtewt, sending messages to different recipients, replying and checking that every conversation is ordered and displayed. That should be good enough.

**Comment by newtewt** [comment]
@vimemo  noticing when sending a message from the admin console to a recipients contact number it is not merging the messages in the app. Not sure if this is an issue caused by this ticket or by the new message sending. 

Recreation Steps:

Log in to beta.dev 
Navigate to messages.
View the messages for Nick with contact number of +16148885555
Navigate to Admin Console >  Message Test
Enter a message Body
Enter a from of +16148885555
Submit
Return to messages and you'll see a new message added to a new conversation with +16148885555 instead of being merged into the existing conversation with Nick

**Comment by vimemo** [comment]
@newtewt, the issue described in your previous comment does not relate to this ticket. It seems that the behavior you described relates to the transition -update_clinics- not being configured/loaded in that environment. Thanks you @garethbowen for pointing this out. @garethbowen, Is this a required transition? Should it be?

**Comment by newtewt** [comment]
Enabling the update_clinics setting made the expected change. 

Closing this as messages are sending and receiving as expected.

**Comment by garethbowen** [comment]
It should not be a required transition IMO because not all projects do messaging.
